openedx/core/djangoapps/user_api/views.py

```diff
@@ -933,12 +934,23 @@ def _apply_third_party_auth_overrides(self, request, form_desc):
                         running_pipeline.get('kwargs')
                     )
 
+                    # When the TPA Provider is configured to skip the registration form and we are in an
+                    # enterprise context, we need to hide all fields except for terms of service and
+                    # ensure that the user explicitly checks that field.
+                    hide_registration_fields_except_tos = (current_provider.skip_registration_form and
+                                                           enterprise_customer_for_request(request))
+
                     for field_name in self.DEFAULT_FIELDS + self.EXTRA_FIELDS:
                         if field_name in field_overrides:
                             form_desc.override_field_properties(
                                 field_name, default=field_overrides[field_name]
                             )
 
+                            if field_name != 'terms_of_service' and hide_registration_fields_except_tos:
+                                form_desc.override_field_properties(
+                                    field_name, field_type="hidden"
+                                )
```

Comment:
I think we also need to not hide the `honor_code` field as well. It looks like that is the name we use when we have a combined TOS/Honor Code field configured.
```
for field_name in self.DEFAULT_FIELDS + self.EXTRA_FIELDS:
    if field_name in field_overrides:
        if current_provider.skip_registration_form and field_name not in ['honor_code', 'terms_of_service']:
            field_override = field_overrides[field_name]
            field_override['field_type'] = 'hidden'
            form_desc.override_field_properties(field_name, default=field_override)
```